lib/utils/config.ts

```diff
@@ -92,14 +92,19 @@ export async function generateApplicationConfig(
 		version: string;
 		deviceType?: string;
 		appUpdatePollInterval?: number;
+		'provisioning-key-name'?: string;
```

Comment:
Yea I've seen that double key creation before during the changes for open-balena, so it predates me as well.

For this PR, I would be fine if the provisioning api key generated by `generateBaseConfig()` (which isn't embedded anywhere) didn't have a name, so that there is no confusion.
Just omit the `provisioningKeyName` property from the options passed to the `generateBaseConfig()`, and add it to `appllication.generateApiKey()` (which btw we can also inline, since it's 2 lines of actual code).

It seems though that at this point the whole `generateApplicationConfig()` function can just be deleted in favor of a simple call to `generateBaseConfig()`, since the API endpoint that it reaches seems to already include the exact same checks :tada: 
See: https://github.com/balena-io/open-balena-api/blob/86f873371c00ddf4dbadcb532cf3b471e22013ba/src/features/device-config/device-config.ts#L50-L71

Reply:
Here is a PR that does exactly that.
Would love some help testing it though.